cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -961,3 +972,95 @@ func (ht *hcsTask) ProcessorInfo(ctx context.Context) (*processorInfo, error) {
 		count: ht.host.ProcessorCount(),
 	}, nil
 }
+
+// Add mount as vSMB share to the UVM at the given destination path
+func (ht *hcsTask) addMountToUVM(ctx context.Context, src string, dst string, isRO bool) (string, error) {
+	options := ht.host.DefaultVSMBOptions(isRO)
+	vsmbShare, err := ht.host.AddVSMB(ctx, src, options)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to add mount as vSMB share to UVM")
+	}
+	defer func() {
+		if err != nil {
+			_ = vsmbShare.Release(ctx)
+		}
+	}()
+
+	sharePath, err := ht.host.GetVSMBUvmPath(ctx, src, isRO)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get vsmb path")
+	}
+	return sharePath, nil
+}
+
+func (ht *hcsTask) requestAddContainerMount(ctx context.Context, resourcePath string, settings interface{}) error {
+	modification := &hcsschema.ModifySettingRequest{
+		ResourcePath: resourcePath,
+		RequestType:  guestrequest.RequestTypeAdd,
+		Settings:     settings,
+	}
+	return ht.c.Modify(ctx, modification)
+}
+
+func isMountTypeSupported(mountType string) bool {
+	// currently we only support mounting of host volumes/directories
+	return mountType == ""
```

Comment:
As Kathryn said, I think we should explicitly define the `mountType` value even if there is only one that we support right now.

Reply:
Are you suggesting we list the other mount types and error out explicitly for them? Not very sure I understand

Reply:
Are you referring to this conversation - https://github.com/microsoft/hcsshim/pull/1918#discussion_r1340497602 ? I am adding mountType constants. Upstream CRI does not set the mountType and passes an "". In internal CRI, the default mountType set is "" when it doesn't fall under any of the other scsi types or sandbox:// types hcsshim supports. Will add this  explicitly.

Reply:
updated the mount types to constant types